lib/base-compiler.js

```diff
@@ -215,8 +235,9 @@ class BaseCompiler {
             options = options.concat(this.compiler.optArg);
         }
 
+        const libIncludes = this.getIncludeArguments(libraries);
         userOptions = this.filterUserOptions(userOptions);
-        return options.concat(userOptions || []).concat([this.filename(inputFilename)]);
+        return libIncludes.concat(options).concat(userOptions || []).concat([this.filename(inputFilename)]);
```

Comment:
Some compilers have include paths specified in their options, and this would result in library include paths coming first in include ordering, which could be a potentially breaking change.

Additionally, this can all be condensed into a single `concat` call.

I tried the following locally with success:
```diff
diff --git a/lib/base-compiler.js b/lib/base-compiler.js
index eb429780..90c86654 100644
--- a/lib/base-compiler.js
+++ b/lib/base-compiler.js
@@ -236,8 +236,8 @@ class BaseCompiler {
         }

         const libIncludes = this.getIncludeArguments(libraries);
-        userOptions = this.filterUserOptions(userOptions);
-        return libIncludes.concat(options).concat(userOptions || []).concat([this.filename(inputFilename)]);
+        userOptions = this.filterUserOptions(userOptions) || [];
+        return options.concat(libIncludes, userOptions, [this.filename(inputFilename)]);
     }
```